crates/matrix-sdk/src/room/common.rs

```diff
@@ -1024,6 +1024,44 @@ impl Common {
         let via = self.route().await?;
         Ok(self.room_id().matrix_uri(via.iter().map(Deref::deref), join))
     }
+
+    /// Get a `matrix.to` permalink to an event in this room.
+    ///
+    /// We try to use the synced members in the room for [routing] the room ID.
+    ///
+    /// # Arguments
+    ///
+    /// * `event_id` - The ID of the event.
+    ///
+    /// [routing]: https://spec.matrix.org/v1.3/appendices/#routing
+    pub async fn matrix_to_event_permalink(
```

Comment:
These methods don't actually ensure that the referenced event is part of the room, no? We only collect a couple of relevant servers and the URL points to the room and event via those servers.

I think we should mention this in the docs, the caller needs to ensure that this event belongs to this room.